src/character.cpp

```diff
@@ -1936,6 +1936,7 @@ bool Character::pour_into( item &container, item &liquid )
         auto qty = std::min( liquid.charges, container.ammo_capacity() - container.ammo_remaining() );
         liquid.charges -= qty;
         container.ammo_set( liquid.typeId(), container.ammo_remaining() + qty );
+        container.on_contents_changed();
```

Comment:
> automatic access (visiting) is explicitly prevented.

Good. This was one of the design goals of `visitable` in that we can specify such rules once only.

> At the moment it needs special casing for all relevant cases.

Everything outside of `item` should use `visitable` which already includes intrinsic support for nested containers. Within `item` it's not so easy to enforce and I'm not sure what design would be best